src/Server/src/Abstractions/Model/Client.cs

```diff
@@ -71,7 +71,7 @@ public class Client : ITenantResource
         /// <summary>
         /// Specifies the allowed grant types (legal combinations of AuthorizationCode, Implicit, Hybrid, ResourceOwner, ClientCredentials).
         /// </summary>
-        public ICollection<string>? AllowedGrantTypes { get; set; } = new HashSet<string>();
+        public ICollection<string> AllowedGrantTypes { get; set; } = new HashSet<string>();
```

Comment:
This will lead to an issue in the future. 
If the intend is to have AllowedGrantTypes as a set, the type should be a set. 
This you can assing any collection to AllowedGrantTypes and in the end the set semantics are not given

e.g. 
Correct behaviour
```csharp
ClientScope x = new();
Client a = new();
a.AllowedGrantTypes.Add(x);
a.AllowedGrantTypes.Add(x);
Console.WriteLine(a.AllowedGrantTypes.Count); // 1
```

Correct behaviour
```csharp
ClientScope x = new();
Client a = new();
a.AllowedGrantTypes = a.AllowedGrantTypes.Where(x =>....).ToList();
a.AllowedGrantTypes.Add(x);
a.AllowedGrantTypes.Add(x);
Console.WriteLine(a.AllowedGrantTypes.Count); // 2
```

Reply:
I don't understand what you mean, here I just removed the nullable because the prop it's already initialized